### DISCUSSION TITLE
userAgent prints undefined when asked about the type information in next15

### DISCUSSION THREAD

**Discussion by akshayysharma** [vercel/next.js#74425] [Help]
### Summary

Hi, can someone explain the scenarios where the type string might be undefined? I am sharing the link below, which provides information about the 'type,' 'browser,' etc., in the userAgent for Next.js.

https://nextjs.org/docs/pages/api-reference/functions/userAgent.

**Comment by icyJoseph** [comment]
This SO answer: https://stackoverflow.com/a/24274846

Shows that including the user-agent header is a `recommendation`. But in practicals terms, I can just make a request to my server, and not include any user-agent. 

See this Node.js echo server:

<img width="1478" alt="Screenshot 2024-12-30 at 19 08 47" src="https://github.com/user-attachments/assets/18c3ee07-6b99-4e85-866e-7d6481933084" />

In the first cURL request, cURL, included a user agent header.

In the second cURL request, we sent nothing. 

Notice that the echo server, then shows no user-agent present, at all.
